apps/web/app/[locale]/(app)/dashboard/[organization]/[list]/table.tsx

```diff
@@ -2,25 +2,77 @@
 
 import { DataTable } from '@/components/data-table/data-table';
 import { Attribute, Contact } from '@prisma/client';
-import { getContactColumns } from './columns';
+import { TableProps, getContactColumns } from './columns';
 import { CustomerAttributes } from '@/lib/types/data-columns';
+import { HeaderActions } from './header-actions';
+import { useToast } from '@/components/ui/use-toast';
+import { Table } from '@tanstack/react-table';
+import { useRouter } from 'next/navigation';
 
 type ContactTableProps = {
   contacts: Contact[];
   attributes: Attribute[];
+  organization: string;
+  list: string;
 };
 
-export const ContactTable = ({ contacts, attributes }: ContactTableProps) => {
-  const columns = getContactColumns(attributes);
+export const ContactTable = ({
+  contacts,
+  attributes,
+  organization,
+  list,
+}: ContactTableProps) => {
+  const { toast } = useToast();
+  const router = useRouter();
+
+  const deleteContacts = async (table: Table<TableProps>) => {
+    // get all selected contacts
+    const selectedRows = table.getFilteredSelectedRowModel().rows;
+    const contactIds = selectedRows.map(row => row.original.id);
+
+    // send request to delete contacts
+    const response = await fetch('/api/contacts', {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ contactIds }),
+    });
```

Comment:
This is not something we'd need to add in this PR, but what do you think about soft deletes for contacts?

Reply:
Mhm, I'm not sure. Could you elaborate more on how you want to soft delete? So, setting a flag or setting values/attributes to null / ****? We need to anonymize or completely delete the contacts from a data privacy view, right? Plus, the email is currently a unique key, so setting it to null or soft deleting it would lead to errors when importing the same contact again.
Or do you mean removing it from this particular list with soft delete? So it stays in a "stage" database and could be imported into another list?

Reply:
fair points. I haven't considered the data privacy aspect, frankly. My idea was to add a `deleted` boolean that gets set to false. I think there are scenarios were such a functionality (i.e. undoing changes) can be helpful.

but we can think about adding it later on 👍